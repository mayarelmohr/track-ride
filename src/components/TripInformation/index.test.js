import React from "react";
import TripInfo from "./index";

import { render } from "@testing-library/react";

test("Shows time and distance correctly", () => {
  const time = 120000;
  const distance = 45;
  const { getByTestId } = render(<TripInfo distance={distance} time={time} />);
  const tripInfoElement = getByTestId("trip-info");
  expect(tripInfoElement).toHaveTextContent(distance);
  const expectedTime = 120000 / 1000;
  expect(tripInfoElement).toHaveTextContent(expectedTime);
});
