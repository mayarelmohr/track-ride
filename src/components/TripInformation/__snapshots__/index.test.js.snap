// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add snapshot for trip information  1`] = `
<DocumentFragment>
  <section
    data-testid="trip-info"
  >
    <h2>
      Trip Information
    </h2>
    <p>
      Trip Duration: NaN seconds
    </p>
    <div
      class="wrapper"
    >
      <div
        class="info"
      >
        <div
          class="image"
        >
          <img
            alt="Robert"
            src="https://robohash.org/1234?size=50x50"
          />
        </div>
        <div
          class="car"
        >
          <img
            alt="car mode"
            src="./images/van.svg"
          />
        </div>
        <div>
          <p
            class="highlighted"
          >
            Herbert Patton
            <span
              class="rate"
            >
              <span
                aria-label="rating"
                role="img"
              >
                ‚≠êÔ∏è
              </span>
              5.0
            </span>
          </p>
          <p>
            Toyato HiAce - GA 2770
          </p>
        </div>
      </div>
      <div
        class="destination"
      >
        <span>
           New Cairo
        </span>
        <span>
           Nasr City
        </span>
      </div>
      <dl
        class="about"
      >
        <div>
          <dt>
            <span
              aria-label="location"
              role="img"
            >
              üìç
            </span>
            Trip Distance:
          </dt>
          <dd>
            undefined KM
          </dd>
        </div>
        <div>
          <dt>
            <span
              aria-label="fare"
              role="img"
            >
              üíµ
            </span>
            Trip Fare:
          </dt>
          <dd>
            45 EGP
          </dd>
        </div>
      </dl>
    </div>
  </section>
</DocumentFragment>
`;
